lib/SimpleSAML/Metadata/SAMLParser.php

```diff
@@ -1329,9 +1320,9 @@ private static function parseGenericEndpoint(EndpointType $element)
      *
      * @return array Array of parsed endpoints.
      */
-    private static function extractEndpoints(array $endpoints)
+    private static function extractEndpoints(array $endpoints): array
     {
-        return array_map(['self', 'parseGenericEndpoint'], $endpoints);
+        return array_map('self::parseGenericEndpoint', $endpoints);
```

Comment:
Is this correct? Does it work?

Reply:
https://www.php.net/manual/en/language.types.callable.php#language.types.callable.passing
See Example 1, Type 4..  The array-format wasn't accepted by Psalm for some reason (probably because it's pre-PHP5.2 style, see Type 2)

Reply:
I know that's possible and it works, but I was wondering specifically about using `self` instead of the class name. In any case, I just tested and it works perfectly fine 😄